etchosts/etchosts.go

```diff
@@ -119,25 +119,19 @@ func Add(path string, recs []Record) error {
 		return nil
 	}
 
-	f, err := os.Open(path)
-	if err != nil {
-		return err
-	}
-
-	content := bytes.NewBuffer(nil)
-
-	_, err = content.ReadFrom(f)
+	f, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND, 0644)
 	if err != nil {
 		return err
 	}
+	defer f.Close()
 
 	for _, r := range recs {
-		if _, err := r.WriteTo(content); err != nil {
```

Comment:
I think the original idea here was to be able to either write all the records to the file or none.
In theory this change would allow a partial update of the etc/hosts file, in case of `WriteTo` failure.